app/code/Magento/MediaGallerySynchronization/Test/Integration/Model/Filesystem/GetFileInfoTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallerySynchronization\Test\Integration\Model\Filesystem;
+
+use Magento\MediaGallerySynchronization\Model\Filesystem\GetFileInfo;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Integration test for GetFileInfo
+ */
+class GetFileInfoTest extends TestCase
+{
+    /**
+     * @var GetFileInfo
+     */
+    private $getFileInfo;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+        $this->getFileInfo = Bootstrap::getObjectManager()->get(GetFileInfo::class);
+    }
+
+    /**
+     * @dataProvider filesProvider
+     * @param string $file
+     */
+    public function testExecute(
+        string $file
+    ): void {
+
+        $path = $this->getImageFilePath($file);
+
+        $fileInfo = $this->getFileInfo->execute($path);
+        $this->assertNotEmpty($fileInfo->getPath());
```

Comment:
Please add more strict verification `$this->assertEquals($path, $fileInfo->getPath());`, `$this->assertEquals($mtime, $fileInfo-> getMTime());`

The expected data can be retrieved from `new \SplFileInfo($path)`